app/assets/javascripts/Components/dashboard.jsx

```diff
@@ -30,16 +34,66 @@ class Dashboard extends React.Component {
             backgroundColor: 'rgb(75, 192, 192)',
           },
         ],
-      }
+      },
+      gradeEntryFormChart2: {}
     };
   }
 
+  GradeEntryFormColumnBreakdownUpdateData = (data) => {
+    // Helper function to update the chart state
+    let label_list = Array();
+    for(let i = 0; i < 105; i+=5) {
```

Comment:
Most of this processing should be done in the controller method. That is, that method's JSON response should already be in the form of a list of objects with `label` and `data` keys. The only thing you should add in this method is the `backgroundColor` key.